### PR TITLE
Mieubrisse/register files endpoints

### PR DESCRIPTION
**Pull request by mieubrisse**
Reason this is necessary: currently, there is only a single Docker volume shared across all tests. The initializer populates that volume with the static files & files artifacts, and then they're available to all the API containers & tests in that execution. 

We need to change this so that static files & files artifacts can be registered without an initializer container, since Kurtosis Interactive wont' have an initializer container.

### CODE REVIEW

File: golang/lib/networks/network_context.go [hunk 1]
```diff
@@ -38,11 +38,9 @@ Creates a new NetworkContext object with the given parameters.
 */
 func NewNetworkContext(
 	client kurtosis_core_rpc_api_bindings.ApiContainerServiceClient,
-	filesArtifactUrls map[services.FilesArtifactID]string,
```
Comment (mieubrisse) [inline golang/lib/networks/network_context.go:41]: With this gone, NetworkContext is now truly stateless!!!

File: golang/lib/networks/network_context.go [hunk 2]
```diff
@@ -124,9 +190,6 @@ func (networkCtx *NetworkContext) AddServiceToPartition(
 	}
 	serviceIpAddr := registerServiceResp.IpAddr
 
-	if err != nil {
```
Comment (mieubrisse) [inline golang/lib/networks/network_context.go:127]: This was error-checking for a new-removed line; it's no longer needed

File: golang/lib/networks/network_context.go (hunk 1 above)
Comment (leoporoli) [reply golang/lib/networks/network_context.go:41]: 👍

File: golang/lib/networks/network_context.go [hunk 3]
```diff
@@ -80,6 +79,74 @@ func (networkCtx *NetworkContext) GetLambdaContext(lambdaId modules.LambdaID) (*
...
+
+		srcFp, err := os.Open(srcAbsFilepath)
+		if err != nil {
+			return stacktrace.Propagate(err, "An error occurred opening static file '%v' source file '%v' for reading", staticFileId, srcAbsFilepath)
+		}
+		defer srcFp.Close()
+
+		destFp, err := os.Create(destAbsFilepath)
+		if err != nil {
+			return stacktrace.Propagate(err, "An error occurred opening static file '%v' destination file '%v' for writing", staticFileId, destAbsFilepath)
```
Comment (leoporoli) [inline golang/lib/networks/network_context.go:125]: This log test says "opening" I guess it should be "creating"

Comment (leoporoli) [reply golang/lib/networks/network_context.go:125]: mmm Now I'm understanding, it says opening in destination for writing,  ok it makes sense, don't listen to me

**Review by leoporoli — APPROVED** [review]

**Pull request merged**